### DISCUSSION TITLE
RetrievalQAWithSourcesChain Implementation

### DISCUSSION THREAD

**Discussion by cycloner2020** [FlowiseAI/Flowise#188] [Ideas]
I got to say this blows LangFlow away! Are there future plans to implement RetrievalQAWithSourcesChain in the near future by chance??

**Comment by HenryHengZJ** [comment]
there is! https://github.com/FlowiseAI/Flowise/issues/116

but need to understand what is the use cases there? are you looking to return something like this?:
![image](https://github.com/FlowiseAI/Flowise/assets/26460777/e071bad9-f165-4e73-b196-d1ea44ca6681)

or are you looking to connect/route the document chunks returned by `RetrievalQAWithSourcesChain` to other components?

**Comment by cycloner2020** [comment]
Ah, ok. Great! I was looking to use to display where the answer to a question was sourced from when using Langchain doc loaders. Here is the example that Langchain uses for RetrievalQAWithSourceChain.


from langchain.chains import RetrievalQAWithSourcesChain

qa_with_sources = RetrievalQAWithSourcesChain.from_chain_type(
    llm=llm,
    chain_type="stuff",
    retriever=vectorstore.as_retriever()
)
qa_with_sources(query)
{'question': 'who was Benito Mussolini?',
 'answer': 'Benito Mussolini was an Italian politician and journalist who was the Prime Minister of Italy from 1922 until 1943. He was the leader of the National Fascist Party and invented the ideology of Fascism. He became dictator of Italy by the end of 1927 and was friends with German dictator Adolf Hitler. Mussolini attacked Greece and failed to conquer it. He was removed by the Great Council of Fascism in 1943 and was executed by a partisan on April 28, 1945. After the war, several Neo-Fascist movements have had success in Italy, the most important being the Movimento Sociale Italiano. His granddaughter Alessandra Mussolini has outspoken views similar to Fascism. \n',
 'sources': 'https://simple.wikipedia.org/wiki/Benito%20Mussolini, https://simple.wikipedia.org/wiki/Fascism'}

**Reply by HenryHengZJ** [reply]
Will have a look at LangChainJS to see if we have `RetrievalQAWithSourcesChain`, to my knowledge there is a `returnSourceDocument` option - https://github.com/FlowiseAI/Flowise/issues/116 but it does not return the same way as `RetrievalQAWithSourcesChain`

**Comment by cycloner2020** [comment]
Or perhaps is there a way to get the source of answer from the response metadata and present to user?
